src/components/header/search/search.module.css

```diff
@@ -1,143 +1,140 @@
 .search {
-  position: relative;
-}
-
-.searchIcon,
-.searchInput,
-.searchInput::placeholder {
-  transition: all 0.2s ease-in-out;
-}
-
-.backgroundTransparent {
-  .searchInput {
+  > :global.DocSearch-Button {
```

Comment:
You might be able to remove `> ` from this and the other selectors in here so the styles are a bit less coupled to the exact HTML structure.

That'll make us a bit more resilient to changes in the Algolia component. For example, since `>` is the direct child selector, an update in the Algolia component that adds another element between `.search` and `.DocSearch-Button` will break our styles:

```html
<!-- Works -->
<div class="search">
   <button class="DocSearch-Button">Search</button>
</div>

<!-- Breaks -->
<div class="search">
   <div class="DocSearch-Button-Wrapper">
     <button class="DocSearch-Button">Search</button>
   </div>
</div>
```

A similar suggestion is to reduce the nesting to not rely on the HTML structure if we don't have to. For example, `.DocSearch-Button-Container` and `.DocSearch-Button-Placeholder` can be siblings at the same level instead of parent and child. Same with `.DocSearch-SearchBar` and `.DocSearch-Form`, etc.

Basically unnest as much as you can and remove `>` until stuff breaks :P